crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -134,36 +164,28 @@ impl Display for ChainId {
 }
 
 /// Parses a string intended to represent a `ChainId` and, if successful,
-/// returns a tuple containing the chain name and revision number.
-fn parse_chain_id_string(chain_id_str: &str) -> Result<(&str, u64), IdentifierError> {
-    let (name, rev_number_str) = match chain_id_str.rsplit_once('-') {
-        Some((name, rev_number_str)) => (name, rev_number_str),
-        None => {
-            return Err(IdentifierError::InvalidCharacter {
-                id: chain_id_str.to_string(),
-            })
-        }
-    };
+/// returns a tuple containing the chain name and (optional) revision number.
+fn parse_chain_id_string(chain_id_str: &str) -> Result<(&str, Option<u64>), IdentifierError> {
```

Comment:
Here, we still can stick with the `(&str, u64)` type, and in case of parsing failure, it defaults to 0. The revision number is always mandatory.